### PR TITLE
Add operator position to list, map and call expressions

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose

- Add operator position to list and map expressions
- Add operator position to call (FunctionCallExpr, MethodCallExpr, CheckingExpr) AST expressions
- Add position to corresponding BIR instructions

 Related to #604, #605

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

## Related PRs
combines #653, #655

### CODE REVIEW

File: compiler/modules/bir/verify.bal [hunk 1]
```diff
@@ -151,10 +151,10 @@ function verifyCall(VerifyContext vc, CallInsn insn) returns err:Semantic? {
     if nSuppliedArgs != nExpectedArgs {
         string name = vc.symbolToString(func.symbol);
         if nSuppliedArgs < nExpectedArgs {
-            return vc.err(`too few arguments for call to function ${name}`);
+            return vc.err(`too few arguments for call to function ${name}`, insn.pos);
         }
         else {
-            return vc.err(`too many arguments for call to function ${name}`);
+            return vc.err(`too many arguments for call to function ${name}`, insn.pos);
```
Comment (jclark) [inline compiler/modules/bir/verify.bal:157]: The position needs to be passed through to and used in verifyOperandType, both here and elsewhere: operandTypeErr needs a Position? argument (which will change to a Position)

File: compiler/modules/front.syntax/parseStmt.bal [hunk 2]
```diff
@@ -218,21 +217,24 @@ function finishCallStmt(Tokenizer tok, CallStmt expr, Position startPos) returns
     return stmt;
 }
 
+// call after ingesting checkingKeyword startPos set to position of checkingKeyword
```
Comment (jclark) [inline compiler/modules/front.syntax/parseStmt.bal]: This comment is not grammatical English. Please start comments that are on a line by themselves with a capital letter,

File: compiler/modules/front.syntax/parseStmt.bal [hunk 3]
```diff
@@ -218,21 +217,24 @@ function finishCallStmt(Tokenizer tok, CallStmt expr, Position startPos) returns
...
 function finishCheckingCallStmt(Tokenizer tok, CheckingKeyword checkingKeyword, Position startPos) returns CallStmt|err:Syntax {
     Token? t = tok.current();
     if t is "check"|"checkpanic" {
```
Comment (jclark) [inline compiler/modules/front.syntax/parseStmt.bal:223]: I still don't understand this. Maybe you explained it in the old pull request.

**Review by jclark — CHANGES_REQUESTED** [review]

**Comment by jclark** [comment]
> I still don't understand this. Maybe you explained it in the old pull request. 

I looked at the old pull request. You can make this code clearer by renaming as I suggested there.

File: compiler/modules/bir/verify.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/bir/verify.bal:157]: fixed

File: compiler/modules/front.syntax/parseStmt.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/parseStmt.bal]: fixed

File: compiler/modules/front.syntax/parseStmt.bal (hunk 3 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/parseStmt.bal:223]: renamed as suggested in #663

File: compiler/modules/front.syntax/parseStmt.bal [hunk 4]
```diff
@@ -218,21 +217,24 @@ function finishCallStmt(Tokenizer tok, CallStmt expr, Position startPos) returns
     return stmt;
 }
 
-function finishCheckingCallStmt(Tokenizer tok, CheckingKeyword checkingKeyword, Position startPos) returns CallStmt|err:Syntax {
+// This function is called with token after the checkingKeword as the current token
```
Comment (jclark) [inline compiler/modules/front.syntax/parseStmt.bal]: Spelling: `Keword`.  Also `the token`.

This is much clearer.

**Review by jclark — COMMENTED** [review]
Looks good.

File: compiler/modules/front.syntax/parseStmt.bal (hunk 4 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/parseStmt.bal]: fixed

**Review by jclark — APPROVED** [review]

**Pull request merged**